extensions/shared/src/main/java/app/revanced/extension/youtube/sponsorblock/SegmentPlaybackController.java

```diff
@@ -545,6 +543,54 @@ private static void setSegmentCurrentlyPlaying(@Nullable SponsorSegment segment)
         SponsorBlockViewController.showSkipSegmentButton(segment);
     }
 
+    /**
+     * Handles changes in player control visibility and manages the skip segment button accordingly.
+     *
+     * <p>This method is called whenever the visibility state of the player controls changes.
+     * If auto-hide is enabled and there is a currently playing sponsor segment, it will show
+     * the skip segment button when the controls are visible and schedule recursive checks
+     * to hide the button after a defined duration.</p>
+     *
+     * @param visible {@code true} if player controls are visible; {@code false} otherwise
+     */
+    public static void onPlayerControlsVisibilityChanged(boolean visible) {
+        if (!Settings.SB_AUTO_HIDE_SKIP_BUTTON.get() || segmentCurrentlyPlaying == null) {
```

Comment:
Like other functions, it is necessary to first check whether the SponsorBlock is enabled.